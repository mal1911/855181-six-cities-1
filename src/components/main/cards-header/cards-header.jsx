import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {getActiveCityName, getCountResultOffers} from '../../../reducer/offers-data/selectors';

const CardsHeader = (props) => {
  return <React.Fragment>
    <h2 className="visually-hidden">Places</h2>
    <b className="places__found">{props.countResultOffers} places to stay in {props.activeCityName}</b>
  </React.Fragment>;
};

CardsHeader.propTypes = {
  activeCityName: PropTypes.string.isRequired,
  countResultOffers: PropTypes.number.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  activeCityName: getActiveCityName(state),
  countResultOffers: getCountResultOffers(state)
});

export {CardsHeader};
export default connect(mapStateToProps)(CardsHeader);
