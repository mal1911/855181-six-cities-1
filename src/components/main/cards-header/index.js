import CardsHeader from "./cards-header";

export default CardsHeader;
