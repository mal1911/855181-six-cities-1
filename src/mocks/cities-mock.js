export const citiesMock = [
  {
    id: 1,
    name: `Paris`,
  },
  {
    id: 2,
    name: `Cologne`,
  },
  {
    id: 3,
    name: `Brussels`,
  },
  {
    id: 4,
    name: `Amsterdam`,
  },
  {
    id: 5,
    name: `Hamburg`,
  },
  {
    id: 6,
    name: `Dusseldorf`,
  },
];

