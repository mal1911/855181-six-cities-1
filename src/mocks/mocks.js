export const offersData = [
  {
    id: 1,
    isPremium: true,
    city: `Paris`,
    title: `Paris Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`,
    coordinates: [52.3909553943508, 4.85309666406198],
  },
  {
    id: 2,
    isPremium: false,
    city: `Paris`,
    title: `Paris Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`,
    coordinates: [52.369553943508, 4.85309666406198],
  },
  {
    id: 3,
    isPremium: false,
    city: `Paris`,
    title: `Paris Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 4,
    isPremium: false,
    city: `Paris`,
    title: `Paris Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 5,
    isPremium: true,
    city: `Cologne`,
    title: `Cologne Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 6,
    isPremium: false,
    city: `Cologne`,
    title: `Cologne Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 7,
    isPremium: true,
    city: `Brussels`,
    title: `Brussels Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 8,
    isPremium: false,
    city: `Brussels`,
    title: `Brussels Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 9,
    isPremium: true,
    city: `Amsterdam`,
    title: `Amsterdam Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 10,
    isPremium: false,
    city: `Amsterdam`,
    title: `Amsterdam Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 11,
    isPremium: true,
    city: `Hamburg`,
    title: `Hamburg Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 12,
    isPremium: false,
    city: `Hamburg`,
    title: `Hamburg Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 13,
    isPremium: true,
    city: `Dusseldorf`,
    title: `Dusseldorf Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 14,
    isPremium: false,
    city: `Dusseldorf`,
    title: `Dusseldorf Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
];

export const citiesData = [
  {
    name: `Paris`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Cologne`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Brussels`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Hamburg`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Dusseldorf`,
    coordinates: [52.38333, 4.9],
  },
];
