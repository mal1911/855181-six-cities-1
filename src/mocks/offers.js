export const offers = [
  {
    title: `Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`
  },
  {
    title: `Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`
  },
  {
    title: `Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`
  },
  {
    title: `Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`
  },
];
