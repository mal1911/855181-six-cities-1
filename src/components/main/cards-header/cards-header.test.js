import React from "react";
import renderer from "react-test-renderer";
import {CardsHeader} from "./cards-header";
import {offersData} from "../../../mocks/mocks";

it(`CardsHeader correctly renders`, () => {
  const tree = renderer
    .create(<CardsHeader
      activeCityName={offersData[0].city.name}
      countResultOffers={10}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});
