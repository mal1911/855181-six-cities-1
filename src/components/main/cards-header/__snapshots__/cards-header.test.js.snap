// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsHeader correctly renders 1`] = `
Array [
  <h2
    className="visually-hidden"
  >
    Places
  </h2>,
  <b
    className="places__found"
  >
    10
     places to stay in 
    Paris
  </b>,
]
`;
